secboot/secboot_sb.go

```diff
@@ -178,11 +186,38 @@ func UnlockVolumeUsingSealedKeyIfEncrypted(disk disks.Disk, name string, sealedE
 	return res, nil
 }
 
-// UnlockEncryptedVolumeUsingKey unlocks an existing volume using the provided key.
-func UnlockEncryptedVolumeUsingKey(disk disks.Disk, name string, key []byte) (UnlockResult, error) {
+func deviceHasPlainKey(device string) (bool, error) {
+	slots, err := sbListLUKS2ContainerUnlockKeyNames(device)
+	if err != nil {
+		return false, fmt.Errorf("cannot list slots in partition save partition: %w", err)
+	}
+
+	for _, slot := range slots {
+		reader, err := sbNewLUKS2KeyDataReader(device, slot)
+		if err != nil {
+			continue
+		}
+		keyData, err := sbReadKeyData(reader)
+		if err != nil {
+			return false, fmt.Errorf("keyslot %s has an invalid key data: %w", slot, err)
```

Comment:
should this continue rather than error out?

Reply:
This is a bit unexpected to happen. To get to this point you need to have a token that is formatted like a secboot token (so type "ubuntu-fde", with "ubuntu_fde_name" set). This is because `NewLUKS2KeyDataReader` would fail otherwise. The key data part ("ubuntu_fde_data") should be able to be deserialized to a `keyData`. And if this deserialization fails, this is where you get an error from `ReadKeyData`. I think if a user gets to this point, there will be more issues. Though, maybe we should continue, but we should probably give a warning.

Reply:
Changed it.